src/Raven.Server/Utils/CertificateUtils.cs

```diff
@@ -86,20 +86,27 @@ internal static bool CertHasKnownIssuer(X509Certificate2 userCertificate, X509Ce
                 return false;
             }
 
-            if (knownCertChain.ChainElements.Count != userChain.ChainElements.Count)
+            // client certificates (leafs) Public Key pinning hashes must match
+            if (userCertificate.GetPublicKeyPinningHash() != knownCertificate.GetPublicKeyPinningHash())
                 return false;
 
-            for (var i = 0; i < knownCertChain.ChainElements.Count; i++)
+            // support self-signed certs
+            if (knownCertChain.ChainElements.Count == 1 && userChain.ChainElements.Count == 1)
+                return true;
+            
+            // compare issuers pinning hashes starting from top of the chain (CA) since it's least likely to change
+            // chain may have additional elements due to cross-signing, that's why we compare every issuer with each other
+            for (var i = knownCertChain.ChainElements.Count - 1; i > 0; i--)
```

Comment:
shouldn't this be `i >= 0` ?

Reply:
No, first element in chain is the client certificate. We want to check only the issuers.